drainer/pump.go

```diff
@@ -134,6 +138,16 @@ func (p *Pump) PullBinlog(pctx context.Context, last int64) chan MergeItem {
 				continue
 			}
 
+			if isBinlogPurged {
+				// some binlogs have been purged in pump, just print the log and wait to exit by user.
+				pLog.Print(labelBinlogGCed, func() {
+					p.logger.Error("some binlogs have been purged in pump")
```

Comment:
It's better to record pump address or pump ID.

Reply:
`p.logger` already combined pump ID, see https://github.com/pingcap/tidb-binlog/blob/3475d287e4da3bd872b90d5386a9918f6fbd7b72/drainer/pump.go#L71.

Reply:
OK